Use mockpostcode, _.merge

sulthan-ahmed opened the pull request:
this is really @JoeChapman work
* use mockpostcode because the live api has some issues with postcodes such as CR02eu which fail tests
* Use _.merge instead of Object.assign
* run all tests in travis not just the lint

sulthan-ahmed committed:
Use mockpostcode, _.merge

* use mockpostcode because the live api has some issues with postcodes such as CR02eu which fail tests
* Use _.merge instead of Object.assign

lennym commented on functional-tests/lib/app.js:
```diff
@@ -8,6 +8,7 @@ const bodyParser = require('body-parser');
 const partials = require('express-partial-templates');
 const template = require('hof-govuk-template');
 const cookieParser = require('cookie-parser');
+const mockPostcode = require('../../mock-postcode');
```
Any compelling reason why this one file is in the app root?

sulthan-ahmed replied:
Not at all, should I move it to lib?

lennym replied:
Yes please.

I'd also add the whole test directory to a `.npmignore` file to keep the installed package light.

lennym replied:
Note: `./functional-tests/lib` not `./lib`

sulthan-ahmed replied:
thanks @lennym I've updated with now the explicit port number.  Hopefully, you are OK with where I put the config.  I had it outside but since it's related to tests, I put it in the functional-test folder

sulthan-ahmed committed:
Update functional tests- explicit port, use config, move mock-postcode

* Have an explicit port number to use as part of the call to the mock api
* Use a config for the port
* Move mock-postcode

lennym commented on functional-tests/index.js:
```diff
@@ -3,14 +3,15 @@
   before(() => {
-    app = App(require('./apps/default')).listen();
+    app = App(require('./apps/default')).listen(config.port);
     port = app.address().port;
```
You don't really need this port variable anymore since it's hardcoded, but it's not doing any real harm, just a little messier than it needs.

lennym committed:
Remove redundant `port` variable

lennym committed:
Use mocha default directory

Mocha looks in ./test by default so using that directory for tests cuts down on unnecessary config.

lennym committed:
Remove redundant auth config in tests

There's no need to set an auth token when we're talking to local mocks

lennym committed:
Run all tests in travis
